fastlane/Fastfile

```diff
@@ -105,6 +108,63 @@ platform :ios do
     # Remove any previous zipped SDK products
     FileUtils.remove_file("../#{deploy_zip}", force = true)
   end
+
+desc "Sets the SDK version in all files"
+  lane :set_version do
+    UI.important "[ATTENTION] Version number should be MAJOR.MINOR.PATCH, do NOT include the `v` prefix."
+    version = UI.input "Enter version number:"
+
+    # Set MARKETING_VERSION using `versioning` plugin
+    increment_version_number_in_xcodeproj(
+      version_number: version,
+      xcodeproj: project_file_path
+    )
+    UI.message "MARKETING_VERSION updated to #{version}"
+
+    version_bump_podspec(path: podspec_path, version_number: version)
+    UI.message "Podspec version updated to #{version}"
+
+    UI.important "You will now update the Changelog. You will be prompted to add entries for any of the available change types. See https://keepachangelog.com/en/1.0.0/ for format."
+
+    today = DateTime.now.strftime("%Y-%m-%d")
+    changelog_changes = "## [#{version}] - #{today}\n"
+    included_change_type_index = 0 # For keeping track of which change types we include to format empty lines correctly
+
+    change_types.each { |type|
+      input = UI.input "Include entry for `#{type}`? (y/n)"
+      if (input.casecmp?(yes_input))
+
+        if (included_change_type_index > 0)
+          changelog_changes += "\n" # Add an extra line between change type sections
+        end
+
+        included_change_type_index += 1
+        changelog_changes += "### #{type}\n" # Add change type title
+
+        finished = false
+        while(!finished)
+          UI.important "[ATTENTION] Do NOT include the `-` prefix in your entry."
+          entry = UI.input "Type your entry for `#{type}`:"
+          entry = "- " + entry + "\n" # Attach the - prefix
+          changelog_changes += entry # Add line to total changes
+
+          additional_input = UI.input "Any additional entries for `#{type}`?"
+          finished = !additional_input.casecmp?(yes_input) # If additional entries is true, we are NOT finished
+        end
+      end
+    }
+    changelog_changes += "\n" # Add a trailing new line to separate from previous sections
+
+    changelog_lines = File.open(changelog_path).readlines
+    changelog_lines.insert(changelog_insertion_line, changelog_changes) # Insert the changes
+
+    # Write out the Changelog including the additions, line by line
+    File.open(changelog_path, "w+") do |f|
+      changelog_lines.each { |element| f.puts(element) }
+    end
```

Comment:
@shawn-veader I could use your Ruby expertise to review this in case there's a better way to insert changes into a file.

Reply:
Hmm... should we use `prompt` (Fastlane action: https://docs.fastlane.tools/actions/prompt/) for some of these?

Reply:
Looks much better. I'll update with `prompt`.